### PR TITLE
adds CI sanity test steps to github workflow

### PR DESCRIPTION
**Pull request by HiteshRepo**
adds CI sanity test steps to github workflow

#### Does this PR need a docs update or release note?
- [x] :no_entry: No

#### Type of change

- [x] :computer: CI/Deployment

#### Issue(s)
#4754 

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

File: .github/workflows/sanity-test.yaml
Comment (HiteshRepo) [inline .github/workflows/sanity-test.yaml:383]: Converts CLI o/p,

From:
```
Restoring to folder Corso_Test_Sanity_dttm
Generating 4 lists items in Corso_Test_Sanity_dttm

  ID            List                                                         Items  Created               Modified            
  2a1e0850638b  Corso_Test_Sanity_dttm_5dfbe852-ad79-4edc-bce2-5d2b4fc65e97  20     2024-01-15T12:50:21Z  2024-01-15T12:50:21Z
  8e49fd6a0cd6  Corso_Test_Sanity_dttm_2028ba03-8dc3-45cc-b989-f0ba3a38bd94  20     2024-01-15T12:50:29Z  2024-01-15T12:50:29Z
  fdc2f3046708  Corso_Test_Sanity_dttm_1b4be031-53a2-425c-b988-1e401e3ef712  20     2024-01-15T12:50:38Z  2024-01-15T12:50:38Z
  db744ba333e7  Corso_Test_Sanity_dttm_63464c42-7c6a-49b6-9cea-cc350d1a6912  20     2024-01-15T12:50:45Z  2024-01-15T12:50:46Z
```

To:
`Corso_Test_Sanity_dttm_543beed0-aa14-4780-b5b7-cc87fba12f38,Corso_Test_Sanity_dttm_697c0cc1-60da-440c-9e96-65c7ba430098,Corso_Test_Sanity_dttm_297d0b08-13c4-4f13-82cb-000fcb33f47b,Corso_Test_Sanity_dttm_f02f0397-c371-4584-97b2-afe8fa8d6e14`

Comment (HiteshRepo) [inline .github/workflows/sanity-test.yaml:383]: Converts CLI o/p,

From:
```
Restoring to folder Corso_Test_Sanity_dttm
Generating 4 lists items in Corso_Test_Sanity_dt_tm

  ID            List                                                         Items  Created               Modified            
  2a1e0850638b  Corso_Test_Sanity_dt_tm_5dfbe852-ad79-4edc-bce2-5d2b4fc65e97  20     2024-01-15T12:50:21Z  2024-01-15T12:50:21Z
  8e49fd6a0cd6  Corso_Test_Sanity_dt_tm_2028ba03-8dc3-45cc-b989-f0ba3a38bd94  20     2024-01-15T12:50:29Z  2024-01-15T12:50:29Z
  fdc2f3046708  Corso_Test_Sanity_dt_tm_1b4be031-53a2-425c-b988-1e401e3ef712  20     2024-01-15T12:50:38Z  2024-01-15T12:50:38Z
  db744ba333e7  Corso_Test_Sanity_dt_tm_63464c42-7c6a-49b6-9cea-cc350d1a6912  20     2024-01-15T12:50:45Z  2024-01-15T12:50:46Z
```

To:
`Corso_Test_Sanity_dt_tm_543beed0-aa14-4780-b5b7-cc87fba12f38,Corso_Test_Sanity_dt_tm_697c0cc1-60da-440c-9e96-65c7ba430098,Corso_Test_Sanity_dt_tm_297d0b08-13c4-4f13-82cb-000fcb33f47b,Corso_Test_Sanity_dt_tm_f02f0397-c371-4584-97b2-afe8fa8d6e14`

File: .github/workflows/sanity-test.yaml
Comment (HiteshRepo) [inline .github/workflows/sanity-test.yaml:413]: Converts 

`Corso_Test_Sanity_dt_tm_543beed0-aa14-4780-b5b7-cc87fba12f38,Corso_Test_Sanity_dt_tm_697c0cc1-60da-440c-9e96-65c7ba430098,Corso_Test_Sanity_dt_tm_297d0b08-13c4-4f13-82cb-000fcb33f47b,Corso_Test_Sanity_dt_tm_f02f0397-c371-4584-97b2-afe8fa8d6e14`

To `Corso_Test_Sanity_dt_tm`

File: .github/workflows/sanity-test.yaml
Comment (HiteshRepo) [inline .github/workflows/sanity-test.yaml:433]: Converts 

`Corso_Test_Sanity_dt_tm_543beed0-aa14-4780-b5b7-cc87fba12f38,Corso_Test_Sanity_dt_tm_697c0cc1-60da-440c-9e96-65c7ba430098,Corso_Test_Sanity_dt_tm_297d0b08-13c4-4f13-82cb-000fcb33f47b,Corso_Test_Sanity_dt_tm_f02f0397-c371-4584-97b2-afe8fa8d6e14`

To `Corso_Test_Sanity_dt_tm`

File: .github/workflows/sanity-test.yaml
Comment (HiteshRepo) [inline .github/workflows/sanity-test.yaml:391]: Converts 

`Corso_Test_Sanity_dt_tm_543beed0-aa14-4780-b5b7-cc87fba12f38,Corso_Test_Sanity_dt_tm_697c0cc1-60da-440c-9e96-65c7ba430098,Corso_Test_Sanity_dt_tm_297d0b08-13c4-4f13-82cb-000fcb33f47b,Corso_Test_Sanity_dt_tm_f02f0397-c371-4584-97b2-afe8fa8d6e14`

To `Corso_Test_Sanity_dt_tm`

File: .github/workflows/sanity-test.yaml
Comment (HiteshRepo) [inline .github/workflows/sanity-test.yaml:5]: will switch to `main` once PR is approved and ready to merge

File: .github/workflows/sanity-test.yaml
Comment (HiteshRepo) [inline .github/workflows/sanity-test.yaml:535]: will uncomment once PR is approved and ready to merge

File: .github/actions/backup-restore-test/action.yml
Comment (ryanfkeepers) [inline .github/actions/backup-restore-test/action.yml:102]:
```suggestion
        # hence SANITY_TEST_RESTORE_CONTAINER_PREFIX is necessary to differentiate restored from original
```

File: .github/workflows/sanity-test.yaml
Comment (ryanfkeepers) [reply .github/workflows/sanity-test.yaml:385]: can you please add that as a comment on the action, so future maintainers know what it does?

File: .github/workflows/sanity-test.yaml
Comment (ryanfkeepers) [reply .github/workflows/sanity-test.yaml:391]: same here, a comment would be helpful.

File: .github/workflows/sanity-test.yaml
Comment (ryanfkeepers) [reply .github/workflows/sanity-test.yaml:535]: feel free to uncomment it now.  Just make sure you tell warn people why they're seeing failure alerts where they appear.

File: src/cli/restore/sharepoint.go
Comment (ryanfkeepers) [inline src/cli/restore/sharepoint.go:98]: this belongs in a separate PR

File: src/cli/utils/sharepoint.go
Comment (ryanfkeepers) [inline src/cli/utils/sharepoint.go:85]: same here, please keep changes separate.  one PR for releasing support, one PR for testing it.

File: src/cmd/purge/scripts/onedrivePurge.ps1
Comment (ryanfkeepers) [inline src/cmd/purge/scripts/onedrivePurge.ps1:171]: why the full text match?  Shouldn't we only be looking for prefixes?

Comment (HiteshRepo) [reply src/cmd/purge/scripts/onedrivePurge.ps1:171]: while pre-cleanup/purge, wanted to remove lists from previous run with prefix `Corso_Test_Sanity_` as wells as remove restored lists with prefix `Corso_Restore_Corso_Test_Sanity_`.

**Comment by HiteshRepo** [comment]
> PR separation needed. Otherwise just nits.

separate PR for lists enablement: https://github.com/alcionai/corso/pull/5048

File: .github/workflows/sanity-test.yaml
Comment (meain) [inline .github/workflows/sanity-test.yaml:387]: I think you can probably make the docs smaller, ie something like:

"The `awk | tr | sed` is used to get a comma separated list of SharePoint list names."

File: .github/workflows/sanity-test.yaml
Comment (meain) [inline .github/workflows/sanity-test.yaml:408]: Similar to previous, this can also be simplified to:

Get the common prefix for the SP list names.

File: src/cmd/purge/scripts/onedrivePurge.ps1
Comment (meain) [reply src/cmd/purge/scripts/onedrivePurge.ps1:171]: How about we change `Corso_Restore_Corso_Test_Sanity_` to `Corso_Test_Sanity_Restore_Corso_Test_Sanity_`? Ideal would be `Corso_Test_Sanity_Restore_`, but I'm guessing that will be hard.

**Review by meain — CHANGES_REQUESTED** [review]
Missed bringing it up earlier. Now that we have two sharepoint tests switching based on category, we might want to update the `CORSO_LOG_FILE` to include the category if present.

**Review by meain — APPROVED** [review]
LGTM

**Review by ryanfkeepers — APPROVED** [review]

**Pull request merged**
